### DISCUSSION TITLE
Booting from SSD

### DISCUSSION THREAD

**Discussion by josbmw** [tesla-android/issue-tracker#272] [General]
Hi Mike,

The boot time remains an issue. I flashed the USB boot firmware to the Pi and tried to boot Android from SSD to improve boot time, but this does not seem to work.
Any idea what could be the problem?

Jos

**Comment by mikegapinski** [comment]
Even booting from NVME won't make it faster at this point, Tesla Android boots up quicker than other alternatives right now and there are still software things I can improve

**Reply by mikegapinski** [reply]
If you have a decent SD card the boot time is between 40 and 50s now, it was closer to 120s a few months ago; I would not call it an issue at this point

**Reply by josbmw** [reply]
I just timed booting from SD card.
It takes 115 seconds before I see the SSID.
Latest image and the card is a new 120 GB SanDisk Ultra A1.

**Reply by mikegapinski** [reply]
That is way too long... How do you power the pi and cool it? One thing to keep in mind also, there is a huge discrepancy between clients and their scan intevals. My MCU is connected a good 20-30s before my macbook can even see the network. If you use carplay the actual boot time is around 10s faster than the start of wireless carplay connection.

**Reply by josbmw** [reply]
5V/3A power supply. No cooling yet, just running tests.
I use a professional Wifi scanning tool. The network pops up as hidden at boot and only becomes visible after almost 2 minutes.

**Reply by mikegapinski** [reply]
the CPU is at 100% usage for the most time when the pi boots, it limits the cpu frequency to 1/4 of the power if it is overheating.

**Reply by mikegapinski** [reply]
also, if you run Tesla Android without a decent active! cooler it is very slow :)

**Reply by josbmw** [reply]
Ok, I'll get a cooling fan and see what this does. Speed currently is ok once booted I must say when running Waze native.

**Reply by mikegapinski** [reply]
This is the best one on the market: https://www.waveshare.com/ice-tower-fan-v2-for-pi.htm ; this one is decent and there is a dedicated case for it https://botland.com.pl/elementy-montazowe-do-raspberry-pi-4b/17275-wentylator-argon-mini-fan-dla-raspberry-pi-4b-z-wlacznikiem-i-radiatorem-5904422378486.html?cd=19576772303&ad=&kd=&gclid=CjwKCAjwvdajBhBEEiwAeMh1UwNbGgf6bKlieeB-BLoKUjuN42P1OrLb_R5EODlO0IMzxfWFDWVINhoCMNAQAvD_BwE

**Reply by josbmw** [reply]
I was looking at this case, fan looks pretty similar: https://www.amazon.de/GeeekPi-Aluminiumgeh%C3%A4use-Tower-ultraleisem-PWM-L%C3%BCfter/dp/B0BRN4MHMH/ref=sr_1_1?__mk_de_DE=%C3%85M%C3%85%C5%BD%C3%95%C3%91&crid=KBV4E94AEEY2&keywords=B0BRN4MHMH&qid=1685443079&sprefix=b0brn4mhmh%2Caps%2C70&sr=8-1

**Reply by mikegapinski** [reply]
Don't, metal cases create a cage around the board that blocks/degrades Wi-Fi signal

**Reply by josbmw** [reply]
Wifi signal seems to be an issue anyway in the older Tesla's. When will your case be available?

**Reply by mikegapinski** [reply]
you mean the CM4 based hardware with the external antenna? Summer, I am ordering the components for the first batch(150/250) units. I expect better availability In September if all goes well.

The last update made things better for those older cars btw, the increased band size means being able to move more data even when the receiver is meh. I can make it even faster, playing it safe for now. There are 1000s of users, I don't want to make bold moves

**Reply by josbmw** [reply]
I understand. Just googeld a bit for an external antenna, but it's not a walk in the park since there is no connector on the standard Pi 4 board. There are mods on the internet to install one, but I don't have a reflow station.
I'm an old school electronics man (just an old Weller 24V soldering iron) :-).

**Reply by mikegapinski** [reply]
The stock antenna that is located on the PCB is better on 5GHz bands. Just make sure to not block it. People have been using this config for over a year now, you'll be fine

**Reply by josbmw** [reply]
At Farnell no CM4's available... Stock currently remains an issue with the Pi product range.

**Reply by mikegapinski** [reply]
the foundation handles those CM4 orders for b2b clients now. The stores only sell pi4s and they are actually available to order(they are frequently restocked)

**Reply by josbmw** [reply]
Farnell has no stock, no pi4 either. But that's not a problem since I had still one lying around :-).
Checked the link to your case, but [the link](https://botland.com.pl/elementy-montazowe-do-raspberry-pi-4b/17275-wentylator-argon-mini-fan-dla-raspberry-pi-4b-z-wlacznikiem-i-radiatorem-5904422378486.html?cd=19576772303&ad=&kd=&gclid=CjwKCAjwvdajBhBEEiwAeMh1UwNbGgf6bKlieeB-BLoKUjuN42P1OrLb_R5EODlO0IMzxfWFDWVINhoCMNAQAvD_BwE) gives me a fan.

**Reply by mikegapinski** [reply]
I've sent you a link for two different fans; no cases.

If you are looking to get rpi4s look at rpilocator.com

**Reply by josbmw** [reply]
Ah ok, I got "this one is decent and there is a dedicated case for it" but it's only the link to the fan.

Do the other components need no cooling? There are many solutions available with individual heat sinks plus CPU.

**Reply by mikegapinski** [reply]
This is the case that goes with the fan: https://argon40.com/products/argon-poly-raspberry-pi-4-vented-case they are available everywhere, even on amazon

**Reply by josbmw** [reply]
Thanks, Amazon sells it as a kit in Germany, ordered one.

**Reply by josbmw** [reply]
And bought you some coffees :-).
